src/languageHandler.js

```diff
@@ -410,11 +410,12 @@ export function pickBestLanguage(langs) {
 }
 
 function getLangsJson() {
-    return new Promise((resolve, reject) => {
+    return new Promise(async (resolve, reject) => {
         let url;
         try {
             // $webapp is a webpack resolve alias pointing to the output directory, see webpack config
-            url = require('$webapp/i18n/languages.json');
+            url = import('$webapp/i18n/languages.json');
+            if (url.then) url = (await url).default; // webpack resolves to a Module through a promise.
```

Comment:
This should always be a promise? At least, we assume it is any time we create an async dialog so if there's a situation where this doesn't work, the app is already broken in it.

Reply:
`require` and `import` changed behaviour somewhere along the way in the webpack upgrade, though I don't know where (particularly considering we didn't jump that far in terms of versions). `require` doesn't work here anymore at all, and `import` returns a Promise *only* under Webpack - babel's compiled output for the npm package runs it sync instead.

A similar problem is described in a bit more detail here: https://github.com/matrix-org/matrix-react-sdk/commit/18ac2db2ea5c44de4153efbf133b8085ba613075

Reply:
Actually I don't understand why this needs to be a dynamic import at all: it should be entirely static since it's just webpack file-loader? That is to say, can you not just replace this with an import at the top of the module?

I don't think I really understand the two scenarios of webpack / babel compiled output: `@babel/plugin-syntax-dynamic-import` might be adding some promise stuff but fundamentally the babel transpiling can't make the `import` work - only a module bundler can do that?

Reply:
We could probably make it a static import, though the goal here wasn't to make the app better just to get it working again. Do we have any documented reason for why this is the way it is?

As far as I can tell, babel transpiles its own bundler in depending on who is doing the effort. On its own for instance, the `lib` output just leaves the imports as-is (or nearly as-is) for whatever module system down the line, assuming it is going to be Node. When Jest runs the source it packs its own bundler in which behaves differently to webpack and slightly different from Node, breaking the expectation here that it is a promise. When Webpack comes around, it decides that it should be async (for whatever reason) and dumps that on us.

Reply:
Ah right yes, so whatever jest is using to do its bundling is somehow messing it up.

In any case the require dates back to https://github.com/matrix-org/matrix-react-sdk/pull/2658/files which would suggest the only reason it's not a top-level require is probably from when we had `LANGUAGES_FILE`, so I think moving it to a static import at the top of the module should be a much simpler change than changing it to a dynamic import and keeps the app and tests running the same code path.

Reply:
alright, will give it a go and see what breaks (probably everything, because this code is fragile af)

Reply:
So... this file is generated at build time in copy-res at riot-web, not in the react-sdk. So we'd have to work around layer boundaries and refactor the resource build step... I'm vetoing that work for this PR, and for sourcemaps in general - it sounds like an improvement to be made after this lands.

Reply:
Yep, I know, but this is no different from what we do now? Webpack sets the $webapp variable so at bundle time the import gets resolved to the file in the webapp directory, so the import returns a URL that's used for fetching the file at runtime. Until bundle time it should stay as $webapp, whether that bundler is our webpack or whatever jest uses.

I've tried switching the require out for a module-level import on develop and it works fine, so I'd be very surprised if this doesn't work in your branches as it's essentially no different whereas changing `require()` to `import()` is making it work fundamentally differently. How is it breaking for you, or what branch should I be running against to repro?

Reply:
ah, I didn't think to just leave the `$webapp` var in there because it would look weird. Will give it a shot, though it'll still be weird.